source/dem/insertion.cc

```diff
@@ -4,10 +4,34 @@
 #include <sstream>
 
 template <int dim>
-Insertion<dim>::Insertion(const std::vector<std::shared_ptr<Distribution>>
-                            &distribution_object_container)
+Insertion<dim>::Insertion(
+  const std::vector<std::shared_ptr<Distribution>>
+    &distribution_object_container,
+  const parallel::distributed::Triangulation<dim> &triangulation,
+  const DEMSolverParameters<dim>                  &dem_parameters)
+  : clearing_particles(dem_parameters.insertion_info.clearing_particles)
```

Comment:
I think the constructor could be slightly cleaner if you had a parameter clearing with default value false and having a single if inside the constructor instead of parsing the full parameter. Just parse `const bool clearing_particles = false` and have it as a parameter upon calling. Plus, do you need an attribute clearing_particles? Parsing it as a parameter when you need it shouldn't be a big deal.

Reply:
It's not a big deal to have it as a member of the class since in the end it's "the state of the insertion machine". Insertion once built should know if it clears or not.